### ISSUE TITLE
Error: no NSS browser

### ISSUE THREAD

**Issue by michaelmior** [anchordotdev/cli#4]
When trying to use `anchor lcl` with Anchor CLI 0.0.15 installed via Homebrew, I get the following error: `Error: no NSS browser`. I'm on a 2023 MBP with macOS 14.3. My primary web browser is [Arc](https://arc.net/).

**Comment by benburkert** [comment]
@michaelmior thanks for opening the issue, it looks to be caused by the `certutil` command not being present on your system, which is part of the `nss` homebrew package. If you're a homebrew user, a quick fix is to `brew install nss`, otherwise we're working on a fix and will have a release up hopefully soon.

**Comment by michaelmior** [comment]
@benburkert That seems to get me past that stage. But now I get `Error: no NSS database`.

Also, just a note that I saw the upgrade message says

> Run `brew upgrade anchordotdev/tap/anchor` to update to the latest version.

You also have to run `brew update` to fetch the latest formula first.

**Comment by geemus** [comment]
I'm adding update to the recommended command, so in future versions it should look like: `brew update && brew upgrade anchordotdev/tap/anchor`. That should be safe to run and work whether your defaults match mine or not.

Thanks again for your feedback and interest. I'm going to close this issue for now as I think we've resolved things, but do let us know if you run into anything else or have further feedback for us.

**Comment by geemus** [comment]
@michaelmior ah, thanks for the clarification, we will take another look.

**Comment by geemus** [comment]
Ok, we just pushed a new version with what we hope should fix these things for you. v0.0.17 should more gracefully handle the new NSS issue you ran into and future update messages should now be warnings (instead of errors) and suggest using both update and upgrade to avoid that problem. Thanks again for the detailed reports and just let us know if you have any other issues.

**Comment by michaelmior** [comment]
@geemus Thanks! That seems to work now :)

**Comment by geemus** [comment]
@michaelmior awesome. I was pretty confident, but it's always better to have confirmation. Thanks again for working through that with us.

**Comment by sebimarkgraf** [comment]
Hey!
I am having the exact same issue with the current anchor version.

``` sh
anchor version
0.0.20 (darwin/arm64) Commit: c11d743fcc37d2fb95e69e55d30cb835492e7577 BuildDate: 2024-04-04T21:30:23Z
```

The error seems to happen with all ``anchor`` commands. e.g.
``` sh
anchor lcl
| Let's set up lcl.host HTTPS in your local development environment!
|
| lcl.host (made by the team at Anchor) adds HTTPS in a fast and totally free way
| to local applications & services.
# Setup lcl.host HTTPS Local Development Environment `anchor lcl`
    | Once setup finishes, you'll have a secure context in your browsers and local
    | system so you can use HTTPS locally.
# Audit lcl.host HTTPS Local Development Environment `anchor lcl audit`
    | We'll begin by checking your system to determine what you need for your setup.
    - Checked resources on Anchor.dev: need to provision resources.
    * Scanning local and expected CA certificates…⠦
Error: no NSS browser
```

I would be happy to provide more debugging information but did not find any guide in your repository.

**Comment by sebimarkgraf** [comment]
Somehow I didn't think about installing nss and missed the comment earlier in the thread.

Would it make sense to adjust the error message to include the install?

And maybe add a step to the quick start guide?

**Comment by geemus** [comment]
@sebimarkgraf Thanks for the detailed report. Although we have currently set things up to default to using NSS, our intention was for it to gracefully skip it if you don't have nss things installed. Unfortunately, you ran into an edge case of that where we had a bug that meant the detection sometimes was giving false positives.

I believe I've fixed this and I just released v0.21.0 which I'm hoping should get you past the problem without needing to install nss.

Just let me know if you have further issues or feedback to share. Thanks!
